http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -416,4 +423,21 @@ private static boolean hasScope(String scopeParam, String targetScope) {
         }
         return false;
     }
+
+    private String addDeploymentAndQueryScopes(Set<String> allScopes, String paramValue){
+        StringBuilder combinedScopes = new StringBuilder();
+        if (paramValue != null && !paramValue.isEmpty()) {
+            allScopes.addAll(Arrays.asList(paramValue.split("\\s+")));
+        }
+        for (String scope : allScopes) {
+            if (scope != null && !scope.isEmpty()){
+                if (combinedScopes.length() == 0) {
+                    combinedScopes.append(scope);
+                } else if (!hasScope(scope, OIDC_SCOPE)) {//some OpenID providers require openid scope to be added in the beginning
```

Comment:
I think hasScope was previously written to be called with a space separated list of scopes.

Maybe now that our logic has changed we can do something like this:

* Remove the openid scope from the set
* Append the openid scope to the StringBuilder since as you mentioned, some providers require it to be first
* Append everything else in the set

Reply:
Thank you for the comments! 
I have pushed changes addressing the comments up to this.